### ISSUE TITLE
Initialisation of pc_inpainting

### ISSUE THREAD

**Issue by jatentaki** [yang-song/score_sde_pytorch#39]
My question concerns [this line](https://github.com/yang-song/score_sde_pytorch/blob/cb1f359f4aadf0ff9a5e122fe8fffc9451fd6e44/controllable_generation.py#L73):

```python
# Initial sample
x = data * mask + sde.prior_sampling(data.shape).to(data.device) * (1. - mask)
```

Let's assume `data` is normalised to have approx `std=1`. In this case, we're initialising `x` as a tensor that has some parts with `std=1` and some parts with `std=prior_std`, which is certainly out of distribution for the score network. Wouldn't it make more sense to initialise it similarly to the body of `inpaint_update_fn`?

```python
vec_t = torch.ones(data.shape[0], device=data.device) * timesteps[0]
masked_data_mean, std = sde.marginal_prob(data, vec_t)
masked_data = masked_data_mean + torch.randn_like(data) * std[:, None, None, None]
x = masked_data * mask + sde.prior_sampling(data.shape).to(data.device) * (1. - mask)
```

I have tried the modification and visually I can't tell if one is significantly better than the other, but I imagine a more thorough benchmarking could reveal differences in FID.

Original algortithm:
![inpaint-original](https://user-images.githubusercontent.com/22394841/221606474-10871386-5193-44f0-a161-91e4d2fa0ecd.png)

My modification:
![modded](https://user-images.githubusercontent.com/22394841/221606514-be0b0e40-0365-42cd-b3fd-a65fac9cdb8c.png)
